### DISCUSSION TITLE
Bug? LED activation value 0 not allowed

### DISCUSSION THREAD

**Discussion by eurorackworld** [shanteacontrols/OpenDeck#216] [General]
HI,

When playing around with the LEDs I noticed the following; the LEDs can respond on a value for example that is set by an analog control. So if you turn the control around the LED will turn on when the activation value is reached. But the analog controls go from 0-127 while the LED is going from 1-127 and doesn't allow a 0. See screenshot.

<img width="1165" alt="image" src="https://github.com/shanteacontrols/OpenDeck/assets/29734704/a8fed912-54c3-4383-9702-57a8f2cde6ec">

This way the LED cannot turn on when you reach 0 with your analog control. For me a 1 is a bit of useless, and a 0 would be very nice. Could that be changed or am I missing something?

**Comment by paradajz** [comment]
Not a bug, that's how it works at the moment.

**Reply by paradajz** [reply]
I'm curious on why you'd want to turn on the LED on value 0?

**Reply by eurorackworld** [reply]
For some values it is handy to see instantly if the value is 0. For example volume or filtering via a potentiometer, a led can easily confirm that the value is 0 and that filter is fully off. Just a nice handy thing, nothing urgent.

Not sure how this would be for buttons, as I am starting to use buttons with LED more and more, but I think those are often just on or off, although this would mean also 0 or 127 I think, but normally it would be on when it's on 127 but that is not the case in 100% of the occasions of course.

**Reply by eurorackworld** [reply]
just thinking , actually working to add leds around potentiometers, and this would mean i can only turn it of when 0 is reached, but never the other way around.

Not really an issue now, except that case I described on the pot going to 0 but the LED couldn't be turned on. Maybe I can use an inverse value and set this in the VST/synth as well, but that feels like an unnecessary 'fix'

**Reply by paradajz** [reply]
I'll push the change later today.

**Reply by paradajz** [reply]
Pushed.
Firmware: https://github.com/shanteacontrols/OpenDeck/commit/34f151ddd85e5df0f9fabf2f9f398efc1de933db
UI: https://github.com/shanteacontrols/OpenDeckUI/commit/8b0e46d19fa1f1f59deb613c75e9050d62d90020

**Reply by eurorackworld** [reply]
Could it be that now it is allowed to enter a 0 but not saving it..? 

If I enter a 0 and switch to another led or screen and back to the original led the value is back to the previous one.

**Reply by eurorackworld** [reply]
p.s. there is a very good use for value 0, the elektron boxes send values 0 and 1 for mute and unmute. Currently the led of the midi controller using value 1 will work exactly the order way then the elektron boxes do. For example the digitakt is sending a 0 when the led needs to be on;

**Reply by paradajz** [reply]
Cannot reproduce. Which firmware? Which board?

**Reply by eurorackworld** [reply]
argh crap. Firmware 7.4.x , was an older controller I was testing with. The newer one works fine. Sorry..
